strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -0,0 +1,163 @@
+package org.carlspring.strongbox.controllers;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * REST API for browsing storage/repository/filesystem structures.
+ *
+ * @author Guido Grazioli
+ */
+@RestController
+@RequestMapping(path = BrowseController.ROOT_CONTEXT)
+public class BrowseController
+        extends BaseArtifactController
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(BrowseController.class);
+
+    // must be the same as @RequestMapping value on the class definition
+    public final static String ROOT_CONTEXT = "/browse";
+
+    @ApiOperation(value = "List configured storages.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The list was returned."),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> browse()
+    {
+        try
+        {
+            Map<String,SortedSet<String>> storages = new HashMap<>();
+            storages.put("storages", new TreeSet<String>(configurationManager.getConfiguration()
+                    .getStorages().keySet()));
+            return ResponseEntity.ok(new ObjectMapper().writeValueAsString(storages));
+        }
+        catch (Exception e)
+        {
+            logger.error(e.getMessage(), e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(String.format("{ 'error': '%s' }", e.getMessage()));
```

Comment:
@guidograzioli : Could you please re-indent this according to the [coding convention](https://github.com/strongbox/strongbox/wiki/Coding-Convention)? Thanks! :)

Reply:
Sure